imports/plugins/included/taxes-avalara/client/components/avalaraSettingsForm.js

```diff
@@ -0,0 +1,182 @@
+import React, { Component } from "react";
+import { assign, clone } from "lodash";
+import PropTypes from "prop-types";
+import { Form } from "/imports/plugins/core/ui/client/components";
+import { Components } from "@reactioncommerce/reaction-components";
+import { AvalaraPackageConfig } from "../../lib/collections/schemas";
+import { Logs } from "/lib/collections";
+import { Logs as LogSchema } from "/lib/collections/schemas/logs";
+import { i18next } from "/client/api";
+import { Loading } from "/imports/plugins/core/ui/client/components";
+
+/**
+ * @file AvalaraSettingsForm is a React Component used to change Avalara
+ * settings.
+ * @module AvalaraSettingsForm
+ */
+
+
+/**
+ * @method AvalaraSettingsForm
+ * @summary renders a form for updating Avalara settings.
+ * @param {Object} props - some data for use by this component.
+ * @property {Function} handleSubmit - a function for saving new Avalara settings.
+ * @property {Array} hiddenFields - the fields (of the Avalara Package) to hide from the form.
+ * @property {Object} settings - the value of the "settings" field in the Avalara Package.
+ * @property {Object} shownFields - info about the fields the form is to show.
+ * @return {Node} - a React node containing the Avalara settings form.
+ */
+class AvalaraSettingsForm extends Component {
+  /**
+  * @name AvalaraSettingsForm propTypes
+  * @type {propTypes}
+  * @param {Object} props - React PropTypes
+  * @property {Object} fieldsProps - map of field specific properties for avalara settings.
+  * @property {Function} handleSubmit - a function that saves new Avalara settings.
+  * @property {Array} hiddenFields - an array of the Avalara Package's fields
+  * to hide from the settings form.
+  * @property {Object} logFieldsProps - map of field specific properties for logs.
+  * @property {Object} settings - the value of the "settings" field in the Avalara Package.
+  * @property {Object} shownFields - info about the fields of the Avalara Package
+  * that the settings form will allow users to change.
+  * @property {Object} shownLogFields - fields to show from Log schema
+  * @return {Array} React propTypes
+  */
+  static propTypes = {
+    fieldsProp: PropTypes.object,
+    handleSubmit: PropTypes.func,
+    handleTestCredentials: PropTypes.func,
+    hiddenFields: PropTypes.arrayOf(PropTypes.string),
+    logFieldsProp: PropTypes.object,
+    settings: PropTypes.object,
+    shownFields: PropTypes.object,
+    shownLogFields: PropTypes.object
+  };
+  static filteredFields = [ "data.request.data.date", "data.request.data.type" ];
+  static noDataMessage = i18next.t("logGrid.noLogsFound");
+
+  // helper adds a class to every grid row
+  static customRowMetaData = {
+    bodyCssClassName: () =>  {
+      return "log-grid-row";
+    }
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      editingId: undefined,
```

Comment:
What's the thinking around settings this value to explicitly be `undefined`? Not saying it's wrong but it seems weird to define it as undefined

Reply:
Not really, I think this is something that I missed in the cleanup.

Reply:
I do usually define all state properties in the constructor, but I set to `null`, which seems to be the pattern I see most. You don't technically need to do it, but it helps provide a complete list of everything you track in `state` for anyone reading the code.